Libraries/StyleSheet/ColorPropType.js

```diff
@@ -9,26 +9,31 @@
   * @providesModule ColorPropType
   */
 'use strict';
+
 var ReactPropTypes = require('ReactPropTypes');
+
 var tinycolor = require('tinycolor');
 
-var colorValidator = function (props, propName) {
+var ColorPropType = function(props, propName) {
   var selectedColor = props[propName];
-  if (selectedColor === null || selectedColor === undefined || selectedColor.toString().trim() === '') {
-    return new Error(
-      `Invalid argument supplied to ${propName}.Expected a string like #123ADF or 'red'.`
-    );
+  if (selectedColor === undefined) {
+    return;
   }
 
-  if (tinycolor(selectedColor.toString().trim()).isValid()) {
-    return null;
+  if (tinycolor(selectedColor) === null) {
+    return new Error(
+`Invalid color supplied to ${propName}: ${selectedColor}. Valid color formats are
+  - '#f0f' (#rgb)
+  - '#ff00ff' (#rrggbb)
+  - '#ff00ff00' (#aarrggbb)
+  - 'rgb(255, 255, 255)'
+  - 'rgba(255, 255, 255, 1.0)'
+  - 'hsl(360, 100%, 100%)'
+  - 'hsla(360, 100%, 100%, 1.0)'
+  - 'transparent'
+  - 'red'
+  - 0xff00ff00 (0xaarrggbb)`);
```

Comment:
Maybe we shouldn't officially support the raw ints (see discussion above). Specifically raw ints (aarrggbb) will work but you probably shouldn't use them directly.